Package.swift

```diff
@@ -5,7 +5,7 @@ import PackageDescription
 
 let package = Package(
   name: "carton",
-  platforms: [.macOS("10.15.4")],
+  platforms: [.macOS(.v12)],
```

Comment:
Had to bump this because Vapor puts a macOS 12 requirement for `async` routes. Can rewrite those new routes with `EventLoopFuture` instead, but I've been meaning to bump the package manifest to Swift 5.6, which requires macOS 12 to build anyway.

@ephemer please let me know if support for earlier versions of macOS is critical for you.

Reply:
I think the async things are not an essential part of this PR, and the [`@available` declaration in vapor](https://github.com/vapor/vapor/blob/deef482f6f309db01666ac99c8b22b6478ef2be5/Sources/Vapor/Concurrency/RoutesBuilder%2BConcurrency.swift#L4) can be relaxed to 10.15.
I think bumping up the minimum supported OS version is critical for production users, so we should discuss it in another PR.